lib/internal/child_process.js

```diff
@@ -764,9 +764,13 @@ function _validateStdio(stdio, sync) {
         fd: i
       });
     } else if (typeof stdio === 'number' || typeof stdio.fd === 'number') {
```

Comment:
You are probably right. I am going to add a few tests for that function in particular as well